logging/google/cloud/logging/handlers/_helpers.py

```diff
@@ -37,3 +45,53 @@ def format_stackdriver_json(record, message):
     }
 
     return json.dumps(payload)
+
+
+def detect_web_framework():
```

Comment:
Recommendation: Don't try to detect the web framework this way. It's very possible for an app to have two frameworks installed and only be using one.

Instead, see if you can get the header from each possible framework. For example:

```python

get_header_from_flask():
    ...

get_header_from_django():
    ...

get_trace_id():
     checkers = [get_header_from_django, get_header_from_flask]

     for checker in checkers:
         header = checker()
         if header is not None:
             return header

    return header